aws/rust-runtime/aws-config/src/profile/credentials.rs

```diff
@@ -493,4 +501,5 @@ mod test {
     make_test!(retry_on_error);
     make_test!(invalid_config);
     make_test!(region_override);
+    make_test!(credentials_process);
```

Comment:
Will implement.

Reply:
we'll need to figure out how to mock this—probably some sort of "fake command" interface will need to get added to our existing OS shim

Reply:
also, should validate that the credentials process can be used as the source profile for assume role

Reply:
also, should validate that the credentials process can be used as the source profile for assume role

Reply:
👍 I ended up using `echo` for the test in which should be supported on all platforms, but happy to try to sub in command mocking if you prefer.

Ref: [a787b61](https://github.com/awslabs/smithy-rs/pull/1356/commits/a787b61f4da20699afb394e9e43a3305075e7672)